### PR TITLE
♻️ Support absolute import paths

### PR DESCRIPTION
**Pull request by rcebulko**
Adopts an alias configuration very similar to NuxtJS:
- `@/` maps to the project root (`import {dev, user} from '@/src/log'`)
- `@@/` maps to shared core (`import {dict} from '@@/types/object'`)

> NuxtJS provides [4 standard aliases](https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-alias): `@@` and `~~` map to the project root; `@` and `~` map to the core source directory. I'm open to either convention, or bikeshedding something else.

Eliminates long strings of '../../../..' (and by extension makes it easier to refactor and move files since imports won't need updating)

### CODE REVIEW

File: test/unit/core/data-structures/test-priority-queue.js [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import PriorityQueue from '../../../../src/core/data-structures/priority-queue';
+import PriorityQueue from '#core/data-structures/priority-queue';
```
Comment (rcebulko) [inline test/unit/core/data-structures/test-priority-queue.js:17]: Imports like this really show how much cleaner imports can be, and how much easier it'll be to reorganize files when you don't have to update `../../../` to `../../../../../src/core` etc

**Review by samouri — COMMENTED** [review]
The updated import statements look lovely!

Unfortunately, it is unclear to me that this is a net-positive.

1. It adds a level of indirection. Just another item which is strange about AMP's build system.
2. Do you know if "Click to Definition" will still work in IDEs?  This is a huge convenience while working on the codebase and I'd hate to lose it for shorter lines.

**Comment by kristoferbaxter** [comment]
I too am a bit mixed for the same reasons Jake mentioned. Would like to hear others thoughts though.

**Comment by rcebulko** [comment]
> 1. It adds a level of indirection. Just another item which is strange about AMP's build system.
For a more "normal" feel, WDYT about if we used something like `@amp/core` which would feel more like importing from packages
> 2. Do you know if "Click to Definition" will still work in IDEs?  This is a huge convenience while working on the codebase and I'd hate to lose it for shorter lines.
This part I do not know offhand and didn't consider. My editor indexes by symbol without using import paths, but perhaps vscode is different?

File: build-system/tasks/storybook/amp-env/webpack.config.js [hunk 2]
```diff
@@ -20,12 +20,14 @@ module.exports = ({config}) => {
     modules: [
       path.join(__dirname, '../node_modules'),
       path.join(__dirname, '../../../../node_modules'),
+      path.join(__dirname, '../../../..'),
```
Comment (alanorozco) [inline build-system/tasks/storybook/amp-env/webpack.config.js:23]: FMI: Does the Webpack config need to be updated somehow? Can Storybook still get files from `src/`?

**Comment by samouri** [comment]
Writing to confirm that the `jsconfig` did alleviate my concerns with VS Code!

**Pull request closed without merging**